app/src/main/java/com/cmput301w23t40/capturetheqr/PlayerLocation.java

```diff
@@ -1,25 +1,43 @@
 package com.cmput301w23t40.capturetheqr;
 
-import com.google.android.gms.maps.model.LatLng;
+import android.content.Context;
+import android.location.Location;
+
+import com.google.android.gms.location.FusedLocationProviderClient;
+import com.google.android.gms.location.LocationServices;
+import com.google.android.gms.location.Priority;
+import com.google.android.gms.tasks.OnSuccessListener;
 
 import java.util.ArrayList;
 
 /**
  * This class models the location of a player and the QR codes around them
  */
 public class PlayerLocation {
-    interface CallbackNearbyCodes {
+    /**
+     * Callback for when the nearby codes have been processed
+     */
+    public interface CallbackNearbyCodes {
         void onUpdateNearbyCodes();
     }
+
+    /**
+     * Callback for when the current geolocation has been obtained from location services
+     */
+    public interface CallbackLocation {
+        void onUpdateLocation();
+    }
     /**
      * The location of the player
      */
     private QRCode.Geolocation location;
+    private FusedLocationProviderClient locationClient;
     private ArrayList<QRCode> nearbyCodes;
 
-    public PlayerLocation() {
+    public PlayerLocation(Context context) {
```

Comment:
the new parameter makes line 40 in MapActivity an error

Reply:
Fixed